examples/empirical_frechet_mean_uncertainty_sn.py

```diff
@@ -0,0 +1,261 @@
+"""Visualise the uncertainty of the empirical Fréchet mean on the sphere.
+
+The variance of the Fréchet mean FM_n of a sample of n IID random variables
+of variance Var is decreasing more slowly in a sphere than in a Euclidean
+space. This example computes the  modulation factor
+     alpha = Var( FM_n) / ( n * Var)
+for isotropic distributions on hyper-spheres of radius 0 < theta < Pi in
+the sphere S_dim (called here a bubble).
+"""
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+import geomstats.backend as gs
+from geomstats.geometry.hypersphere import Hypersphere
+
+
+def empirical_frechet_var_bubble(n_sample, theta, dim,
+                                 n_expectation=1000):
+    """Variance of the empirical Fréchet mean for a bubble distribution.
+
+    Draw n_samples from a bubble distribution, computes its empirical
+    Fréchet mean and the square distance to the asymptotic mean. This
+    is repeated n_expectation times to compute an approximation of its
+    expectation (i.e. its variance) by sampling.
+
+    The bubble distribution is an isotropic distributions on a Riemannian
+    hyper sub-sphere of radius 0 < theta < Pi around the north pole of the
+    sphere of dimension dim.
+
+    Parameters
+    ----------
+    n_sample: number of samples to draw
+    theta: radius of the bubble distribution
+    dim: dimension of the sphere (embedded in R^{dim+1})
+    n_expectation: number of computations for approximating the expectation
+
+    Returns
+    -------
+    tuple (variance, std-dev on the computed variance)
+    """
+    assert dim > 1, "Dim > 1 needed to draw a uniform sample on sub-sphere"
+    var = []
+    sphere = Hypersphere(dimension=dim)
+    bubble = Hypersphere(dimension=dim - 1)
+
+    # Define north pole
+    north_pole = np.zeros(dim + 1)
+    north_pole[dim] = 1.0
+    for k in range(n_expectation):
```

Comment:
This could be a method from the hypersphere (sampling on a hypersphere of a certain radius). Maybe add a TODO that points towards the refactorization of this code?